### ISSUE TITLE
Remove YoutubeExtractor from the official extractors lib

### ISSUE THREAD

**Issue by skdhg** [Androz2091/discord-player#1726]
Due to the recent shenanigans of youtube-discord music bots, I think it makes sense to exclude youtube support from the official extractors library to support discord/youtube's community guidelines. I am aware that there are many users who are explicitly using youtube source for their music bot. In order to not affect these users, we could move youtube extractor to an external community built package. This also means that other built-in extractors who bridge the source to stream will not work. This could be solved by using soundcloud extractor for stream extraction under the hood.

The extractor api exposed by discord-player can be utilized to create custom extractors as desired. See https://discord-player.js.org/docs/guides/creating-extractor for more info on how to create custom extractors.

Being a maintainer of this lib, I am letting you guys decide what could be the best option for this issue. Please feel free to express your opinions. If majority of us agree upon moving youtube extractor to community package, I am happy to do so. The otherwise is also fine if you guys are okay with it.

Aside from that, I am removing automatic loading of the extractors exposed by the extractors lib from next version (versions >6.2.1). Manually loading the extractors would be better since it lets you pick what you want and stops unnecessary computation.

**Comment by itsauric** [comment]
I wouldn't agree to exclude YouTube support from the extractors library because this is a package that can be used for private use in terms of discord bots and also public use. It will be amongst bot developers to reason their usage of using the package and publicising their discord bots, in other words I would say this is up to Discord and YouTube on how they would sanction the main offenders being bot developers at this time. Removing the YouTube support would not make a whole bunch of difference in regards to community guidelines as well and I am pretty sure a public npm package would not be against the guidelines but rather the bot developer's usage and how they use their discord bots within the guidelines of Discord and YouTube as a well, mainly focusing on YouTube's perspective. 

In conclusion, I fully disagree to "remove" or exclude YouTube support from `discord-player` and for more references too I would assume most music packages also use YouTube support as their main extractor, I may be corrected if wrong.

**Comment by retrouser955** [comment]
That depends. Can YouTube take legal action on open source repos?

**Comment by Sans3108** [comment]
> 

that about sums up what i wanted to say in a more professional manner, yup

**Comment by skdhg** [comment]
alright, based on https://github.com/Androz2091/discord-player/issues/1726#issuecomment-1528788917, I am closing this issue
